cargo-dist/src/init.rs

```diff
@@ -45,6 +45,22 @@ struct MultiDistMetadata {
 
 /// Run 'cargo dist init'
 pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
+    // on ctrl-c,  dialoguer/console will clean up the rest of its
+    // formatting, but the cursor will remain hidden unless we
+    // explicitly go in and show it again
+    // See: https://github.com/console-rs/dialoguer/issues/294
+    let ctrlc_handler = tokio::spawn(async move {
+        tokio::signal::ctrl_c().await.unwrap();
+
+        let term = console::Term::stdout();
+        // Ignore the error here if there is any, this is best effort
+        let _ = term.show_cursor();
+
+        // Immediately re-exit the process with the same
+        // exit code the unhandled ctrl-c would have used
+        std::process::exit(130);
```

Comment:
Does this make sense on windows..?

Reply:
Also process::exit is always a maintenance hazard, since it prevents our own app from running its destructors. I suppose in this case that might be fine / this is the function to do anything like that. Just, noting it for future.

Reply:
Looks like no, Windows uses something extremely different. (I'm seeing `-1073741510` in both `cmd` and powershell, which I guess is `0xc000013a`.) Also looks like the hiding the cursor behaviour is different between `cmd` and powershell...?

Reply:
> Also process::exit is always a maintenance hazard, since it prevents our own app from running its destructors. I suppose in this case that might be fine / this is the function to do anything like that. Just, noting it for future.

Yeah, the thing is I figured this is basically equivalent to what ctrl-c was doing here already.